src/v/kafka/server/errors.h

```diff
@@ -133,6 +133,7 @@ constexpr error_code map_tx_errc(cluster::tx::errc ec) {
     case cluster::tx::errc::partition_not_exists:
     case cluster::tx::errc::not_coordinator:
     case cluster::tx::errc::stale:
+    case cluster::tx::errc::producer_creation_error:
```

Comment:
This error mapping needs more 👀 .. the intent was to map to a retriable error code that works with both idempotent and transactional producers and not_coordinator seems the safest. It provides some natural backoff + retry and meanwhile the producers may get evicted to make space for the incoming request.